src/zenml/model_registries/base_model_registry.py

```diff
@@ -0,0 +1,425 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Base class for all ZenML model registries."""
+
+from abc import ABC, abstractmethod
+from enum import Enum
+from typing import Any, Dict, List, Optional, Type, cast
+
+from pydantic import BaseModel, Field, root_validator
+
+from zenml.enums import StackComponentType
+from zenml.stack import Flavor, StackComponent
+from zenml.stack.stack_component import StackComponentConfig
+
+
+class ModelVersionStage(Enum):
+    """Enum of the possible stages of a registered model."""
+
+    NONE = "None"
+    STAGING = "Staging"
+    PRODUCTION = "Production"
+    ARCHIVED = "Archived"
+
+
+class ModelRegistration(BaseModel):
+    """Base class for all ZenML registered models.
+
+    Model Registration are the top-level entities in the model registry.
+    They serve as a container for all the versions of a model.
+
+    Attributes:
+        name: Name of the registered model.
+        description: Description of the registered model.
+        tags: Tags associated with the registered model.
+    """
+
+    name: str
+    description: Optional[str]
+    tags: Optional[Dict[str, str]] = None
+
+
+class ZenMLModelMetadata(BaseModel):
```

Comment:
Not a huge fan of the `ZenML` prefix in the class name, just `ModelMetadata` maybe?